coralproject/talk
# pre-commit linting

## What does this PR do?

This replaces the [pre-git](https://npmjs.com/package/pre-git) solution with a combination of [husky](https://npmjs.com/package/husky), [@easyops/git-exec-and-restage](https://npmjs.com/package/@easyops/git-exec-and-restage), and [lint-staged](https://npmjs.com/package/lint-staged).

**Benefits:**

- Linting as precommit is _much_ faster, as only changed/staged files are passed to get linted 🐎 
- Adding a commit will not perform the _"warning vomit wall of complaints*_, but can still be retrieved via `yarn lint` 🤢 

**Drawbacks:**

- Testing is not a part of the pre-commit process
  - Testing should still be completed, but we now rely on the fact that tests are completed in CI
  - _Sorta benefit:_ PR's and work that is _in progress_ will not go unlinted because of known failing tests 💇 

Heavily inspired by the work done in the [hexo](https://github.com/hexojs/hexo) package, which implemented the same.

## How do I test this PR?

1. Run `rm -rf .git/hooks/*`
2. Checkout this branch
3. Run `yarn`
4. Make a small change to one file that breaks linting (like calling a function named `thisIsNotAFunction();`
5. Try to commit the file, and notice that the linter catches it and finishes quickly (as it only actually linted that one file)
